node/core/prospective-parachains/src/lib.rs

```diff
@@ -525,14 +526,27 @@ fn answer_prospective_validation_data_request(
 				head_data = Some(required_parent.clone());
 			}
 		}
+		if max_pov_size.is_none() {
+			let contains_ancestor = fragment_tree
+				.scope()
+				.ancestor_by_hash(&request.candidate_relay_parent)
+				.is_some();
+			if contains_ancestor {
+				// We are leaning hard on two assumptions here.
```

Comment:
Both seem sensible, configuration changes are buffered to be applied on new session